### DISCUSSION TITLE
Automerge a group

### DISCUSSION THREAD

**Discussion by jpolo** [renovatebot/renovate#21209] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Wanted end result.

When reading documentation, it is not clear how to mark a defined group as automergeable.

A concrete case would be a group defined in some configuration json file `jest monorepo` + `es-jest` + XXX
=> `automerge: true`

The main challenge seems that Renovate allows to augment a group in different files (groupJestPlusTsJest for example)

### What you tried so far.

Do not know what to do, because no concrete example in documentation...

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment]
For a group PR to be automerged, all updates within the group must be automerge=true. Do you have a concrete example?

**Comment by jpolo** [comment]
Say that I want to setup automerge for `group:ionic-nativeMonorepo` (defined by Renovate presets)

What should I do ?

**Reply by rarkins** [reply]
```
{
  "packageRules": [
    {
      "extends": ["monorepo:ionic-native"],
      "matchUpdateTypes": [
        "patch",
        "minor"
      ],
      "automerge": true
    }
  ]
}
```

**Reply by rarkins** [reply]
See https://docs.renovatebot.com/presets-group/#groupionic-nativemonorepo for reference

**Comment by HonkingGoose** [comment]
> Do not know what to do, because no concrete example in documentation...

I'm fixing this with PR:

- #21223

**Reply by jpolo** [reply]
Thanks for your answer !

If I undestand well your solution, it might partially solve the problem but not entirely, especially on  a slightly more complex example : [groupJestPlusTsJest](https://docs.renovatebot.com/presets-group/#groupjestplustsjest) 

This is "an augmented group".

How to mark this group be mergeable (and possibly all package that would be added to the group) ? If I use your example => that would mean I will have to use `monorepo:jest` but `ts-jest` will not be marked as automergeable, and it will not work properly...

**Reply by rarkins** [reply]
Yes, for an edge like this you will need to copy paste the `matchX` patterns of this group - you cannot extend

**Reply by jpolo** [reply]
I think this would be a good feature request.
By design copying the match conditions is not maintainable because it blocks the whole pull request as soon as a group is extended.

This is a common case for plugin based projects like Babel, jest, eslint, etc.  Updates should most of the time be grouped and groups should be safely extendable as the  community plugins are created...

**Comment by rarkins** [comment, marked as answer]
Added feature request for a `matchBranches` concept to be added to `packageRules`: #21286

PRs welcome

**Reply by jpolo** [reply]
Maybe matchBranch sounds very git related, more something like matchGroup ?
